r/tests/testthat/test-data-type.R

```diff
@@ -390,8 +390,8 @@ test_that("decimal type and validation", {
   expect_error(decimal(4))
   expect_error(decimal(4, "two"), '"scale" must be an integer')
   expect_error(decimal(NA, 2), '"precision" must be an integer')
-  expect_error(decimal(0, 2), "Invalid: Decimal precision out of range: 0")
-  expect_error(decimal(100, 2), "Invalid: Decimal precision out of range: 100")
+  expect_error(decimal(0, 2), "\"precision\" must be greater than 0")
+  expect_error(decimal(100, 2), "\"precision\" must be less than or equal to 38")
```

Comment:
I'm ok adding these, but I'm curious if there's something specifically wrong with the error messages we are seeing from C++ that we're addressing with this re-write of the error message? They seem ~equivalent to me in readability/understandability but I might be too far down this rabbit hole.

_In general_ if the C++ error is understandable, we should defer to that so that we don't need to have a bunch of error handling code duplicated at these two different layers.

Reply:
For example: when/if we implement other decimal precisions we won't have to gate on each of those in the same way.

Reply:
Agree on consistency. I'm not a huge fan of the current error message (surfaced from C++) as it doesn't tell the user what they need to do to prevent the error. `decimal precision out of range` is not informative if we don't tell them anything about said `range`. It reminds me of frustrating account creating experiences when the password isn't "correct", but you don't know what wrong with it and therefore correction requires a lot of trial and error.

Reply:
I agree with your point on the phrasing being tighter @dragosmg .  I know some of the C++ folk might prefer us to raise it with them if we can suggest improvements to the error messages that would be good to implement at the C++ level so it benefits everyone - this potentially could be one of those?

Reply:
Yeah, we should raise a Jira to suggest a better warning. We don't need to hold this PR to wait for that to happen, but maybe a comment with the Jira # so we know to clean it up if/when the jira happens.

Reply:
Done. Restored these to their original state and raised a Jira (first comment) to improve messaging in C++.

Reply:
+ added a comment with the Jira number in the code